src/services/WebAzureSubscriptionProvider.ts

```diff
@@ -94,6 +94,12 @@ class VSCodeAzureSubscriptionProvider extends vscode.Disposable implements Azure
 
         this.allSubscriptions = allSubscriptions;
 
+        // migrate setting values that are just a subscription id to `${tenantId}/${subscriptionId}`
+        const selectedSubscriptionIds = settingUtils.getGlobalSetting<string[] | undefined>('selectedSubscriptions');
```

Comment:
Should we only do this in the constructor? `getSubscriptions` gets called every time `selectSubscriptions` does, which can be pretty often.

Reply:
We need to `await` updating the setting value so I don't think it can be done in the constructor.

Reply:
Ok, I added a `static async Create` method to the class that handles it.